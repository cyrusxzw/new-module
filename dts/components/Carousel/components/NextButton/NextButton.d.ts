import { FC } from 'react';
import type { NextButtonProps } from './NextButton.types';
/**
 * The onClick callback for the NextButton
 * components are provided through react-slick
 */
declare const NextButton: FC<NextButtonProps>;
export { NextButton };
