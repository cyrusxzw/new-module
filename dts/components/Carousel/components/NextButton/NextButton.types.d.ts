import type { MouseEvent } from 'react';
import type { Themes } from '~/types';
declare type NextButtonProps = {
  isActive?: boolean;
  onClick?: (event: MouseEvent) => void;
  theme?: Themes;
};
export type { NextButtonProps };
