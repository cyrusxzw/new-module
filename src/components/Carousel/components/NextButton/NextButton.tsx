import React, { FC } from 'react';
import cx from 'classnames';
import { Button } from '~/components/Button';
import { Icon } from '~/components/Icon';
import type { NextButtonProps } from './NextButton.types';
import styles from './NextButton.module.css';

/**
 * The onClick callback for the NextButton
 * components are provided through react-slick
 */
const NextButton: FC<NextButtonProps> = ({
  isActive = true,
  onClick,
  theme = 'dark',
}) => (
  <Button
    className={cx(styles.base, { [styles.hidden]: !isActive })}
    isInline={true}
    onClick={onClick}
    theme={theme}
    title="next slide"
  >
    <Icon height={16} name="chevron" theme={theme} width={16} />
  </Button>
);

export { NextButton };
