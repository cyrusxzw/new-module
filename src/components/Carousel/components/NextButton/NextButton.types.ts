import type { MouseEvent } from 'react';
import type { Themes } from '~/types';

type NextButtonProps = {
  isActive?: boolean;
  onClick?: (event: MouseEvent) => void;
  theme?: Themes;
};

export type { NextButtonProps };
