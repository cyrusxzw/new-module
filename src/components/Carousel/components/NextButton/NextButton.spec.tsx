import React from 'react';
import { render } from '@testing-library/react';
import { NextButton } from './NextButton';

describe('<Carousel.NextButton />', () => {
  it('should render base component correctly', () => {
    const { container } = render(
      <NextButton
        onClick={() => {
          return;
        }}
      />,
    );

    expect(container).toMatchSnapshot();
  });
});
