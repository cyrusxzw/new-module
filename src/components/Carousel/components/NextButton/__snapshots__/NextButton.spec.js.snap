// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Carousel.NextButton /> should render base component correctly 1`] = `
<div>
  <button
    class="base dark base"
    title="next slide"
    type="button"
  >
    <svg
      aria-labelledby="chevron-1"
      class="base chevron dark"
      data-testid="data-testid-Icon"
      focusable="false"
      height="16"
      role="img"
      style="height: 16px; width: 16px;"
      tabindex="-1"
      viewBox="0 0 50 50"
      width="16"
    >
      <title
        id="chevron-1"
      >
        chevron
      </title>
      <g>
        <polygon
          class="chevron"
          points="25,31.3 4.2,10.5 0.1,14.6 25,39.5 50,14.6 45.9,10.5"
        />
      </g>
    </svg>
  </button>
</div>
`;
